src/main/java/seedu/address/logic/parser/LessonAddCommandParser.java

```diff
@@ -0,0 +1,110 @@
+import static seedu.address.logic.parser.CliSyntax.PREFIX_DATE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_END_TIME;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_HOMEWORK;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_RECURRING;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_START_TIME;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_SUBJECT;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Stream;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.LessonAddCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.lesson.Date;
+import seedu.address.model.lesson.Homework;
+import seedu.address.model.lesson.MakeUpLesson;
+import seedu.address.model.lesson.RecurringLesson;
+import seedu.address.model.lesson.Subject;
+import seedu.address.model.lesson.Time;
+import seedu.address.model.lesson.TimeRange;
+
+/**
+ * Parses input arguments and creates a new LessonAddCommand object.
+ */
+public class LessonAddCommandParser {
+    /**
+     * Parses the given {@code String} of arguments in the context of the LessonAddCommand
+     * and returns a LessonAddCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public LessonAddCommand parse(String args) throws ParseException {
+        requireNonNull(args);
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_RECURRING, PREFIX_DATE, PREFIX_START_TIME,
+                        PREFIX_END_TIME, PREFIX_SUBJECT, PREFIX_HOMEWORK);
+
+        if (!arePrefixesPresent(argMultimap, PREFIX_DATE, PREFIX_START_TIME, PREFIX_END_TIME, PREFIX_SUBJECT)) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, LessonAddCommand.MESSAGE_USAGE));
+        }
+
+        Index index;
+
+        try {
+            index = ParserUtil.parseIndex(argMultimap.getPreamble());
+        } catch (ParseException pe) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, LessonAddCommand.MESSAGE_USAGE), pe);
+        }
+
+        Date date = ParserUtil.parseDate(argMultimap.getValue(PREFIX_DATE).get());
+
+        /* Parse time and check if range is valid.
+        Throws ParseException if either time is invalid
+        or if range is invalid.
+         */
+        Time startTime = ParserUtil.parseTime(argMultimap.getValue(PREFIX_START_TIME).get());
+        Time endTime = ParserUtil.parseTime(argMultimap.getValue(PREFIX_END_TIME).get());
+        TimeRange timeRange = new TimeRange(startTime, endTime);
+        if (!timeRange.isValidTimeRange(startTime, endTime)) {
+            throw new ParseException(TimeRange.MESSAGE_CONSTRAINTS);
+        }
```

Comment:
```suggestion
        if (!TimeRange.isValidTimeRange(startTime, endTime)) {
            throw new ParseException(TimeRange.MESSAGE_CONSTRAINTS);
        }
        TimeRange timeRange = new TimeRange(startTime, endTime);
```

Reply:
currently there's a target invocation error because it returns invalid within TimeRange constructor